subprojects/packagefiles/msgpackc-cxx/meson.build

```diff
@@ -1,27 +1,36 @@
 project(
-    'msgpackc-cxx', 'cpp',
-    version: '6.1.0',
-    meson_version: '>=0.42.0'
+  'msgpack-cxx', 'cpp',
+  version: '6.1.0',
+  meson_version: '>=0.47.0',
 )
 
 args = []
+
+boost_dep = dependency('boost', required: get_option('boost'))
+if not boost_dep.found()
+  args += ['-DMSGPACK_NO_BOOST']
+endif
+
 if get_option('x3_parse')
-    args += ['-DMSGPACK_USE_X3_PARSE']
+  if not boost_dep.found()
+    error('Boost is required for X3 parse')
```

Comment:
now this is interesting. why is there a boot option? Sounds like you could replace get_option('boost') with get_option('x3_parse')

Reply:
No, because there are Boost components enabled by default independent of X3.

Reply:
So boost is mandatory?

Reply:
For most users, yes. These components can be disabled (which is experimental), but that would be a very specific use case (e.g. when someone wants to avoid Boost at all costs).